cmd/ncproxy/ncproxy.go

```diff
@@ -298,7 +359,40 @@ func (s *grpcService) CreateEndpoint(ctx context.Context, req *ncproxygrpc.Creat
 			ID: ep.Id,
 		}, nil
 	case *ncproxygrpc.EndpointSettings_NcproxyEndpoint:
-		return nil, status.Error(codes.Unimplemented, "ncproxy endpoint is not implemented yet")
+		// get the network stored, create endpoint data and store
+		reqEndpoint := req.EndpointSettings.GetNcproxyEndpoint()
+		if reqEndpoint.Name == "" || reqEndpoint.Ipaddress == "" || reqEndpoint.Macaddress == "" || reqEndpoint.NetworkName == "" || reqEndpoint.DeviceDetails == nil {
```

Comment:
If we fall into this case, it's expected that the client always has to pass the PCI details (and for hns this isn't required)?

Reply:
yeah for now we'll require it since the only reason to use the ncproxy networking things is for sriov that doesn't go through hns. In the future we can remove that requirement.

Reply:
Ahh, might be out of the loop now. Didn't we need to set a specific IOV policy through HNS to get it to function though?

Reply:
we only need to set iov policy on an HNS endpoint if that device is managed by hns and we want to use sriov. The ncproxy endpoint and network are for networking that is not managed by hns and therefore we need the pci information so we can assign the device to the uvm manually. Does that clear up the confusion?

Reply:
Ok, makes sense!